openpype/hosts/nuke/api/plugin.py

```diff
@@ -98,10 +93,13 @@ def check_existing_subset(self, subset_name, instance_data):
         """Check if existing subset name already exists."""
         exists = False
         for node in nuke.allNodes(recurseGroups=True):
```

Comment:
Technically the artist could publish a `renderMain` for different assets from the same comp. As such - the check here is too strict. It should actually check whether an instance already exists that will publish the subset for the same asset as you're trying to create now.

Reply:
I agree. I would move the validation to publishing. There is so much ways how to bypass this check and at the same time can cause so much headaches (and slowdown) I would remove it completelly.

Reply:
Yes - a global plugin which validates this is the BEST case scenario since it's valid for basically ALL publishes - unless there are special cases I don't know about where two separate instances should somehow publish and integrate into the same asset+subset from within the same publish.

Here's an old basis for that logic: https://github.com/Colorbleed/colorbleed-config/blob/acre/colorbleed/plugins/global/publish/validate_unique_subsets.py

However as clearly can be seen from that todo it still has the bug that it doesn't check it per asset. I'd be happy to refactor it and set up a PR if we feel this is a worthwhile global plug-in.

Reply:
I like the approach of moving this to publishing. Each instance, should have been able to store the validator disabled value into instance data - so in case user disable it on a instance with different asset name he will still pass the validation phase. 

For now, since the validator is still not in place, lets keep the validation in clreators and remove it in the next PR.